File: framework/src/common/framework/camera.c
#include "GLES/gl.h"
#include "camera.h"
#include "framework/util.h"
void cameraSetActive(Camera* cam)
{
  glMatrixMode(GL_PROJECTION);
  glLoadIdentity();
  glOrthof(cam->posX,  cam->posX + cam->width, cam->posY, cam->posY + cam->height, -1, 1);
  /*float hw = cam->width/2.0;*/
  /*float hh = cam->height/2.0;*/
  /*glOrthof(cam->posX - hw,  cam->posX + hw, cam->posY - hh, cam->posY + hh, -1, 1);*/
  glMatrixMode(GL_MODELVIEW);
}

void cameraInit(Camera* cam, float x,float y, int width, int height)
{
  cam->posX = x;
  cam->posY = y;
  cam->width = width;
  cam->height = height;
}

File: framework/src/lua/camera.lua



framework = framework or {}

framework.Camera = {}

function framework.Camera.new(width, height)
  local M = {}
  local cam = _c_framework.Camera()

  _c_framework.cameraInit(cam, 0, 0, width, height)


  local mt = {}

  mt.__index = function(table, key)
    if key == "x" then
      return cam.posX
    elseif key == "y" then
      return cam.posY
    elseif key == "width" then
      return cam.width 
    elseif key == "height" then
      return cam.height 
    end
  end

  mt.__newindex = function(table, key, value)
    if key == "x" then
      cam.posX = value
    elseif key == "y" then
      cam.posY = value
    elseif key == "width" then
      cam.width = value
    elseif key == "height" then
      cam.height = value
    end
  end

  function M.apply()
    _c_framework.cameraSetActive(cam)
  end

  setmetatable(M, mt)


  return M
end


function framework.Camera.createDefault(width)
  local aspect = framework.Window.getHeight()/framework.Window.getWidth()
  print("Aspect: "..aspect)
  local w = width or (480*2)
  local h = w * aspect
  return framework.Camera.new(w, h)
end
